l3kernel/l3token.dtx

```diff
@@ -2914,6 +2859,57 @@ end
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}{\peek_remove_filler:n}
+% \begin{macro}{\@@_remove_filler:w}
+% \begin{macro}{\@@_remove_filler:}
+% \begin{macro}{\@@_remove_filler_expand:w}
+%   Here we expand the input, removing spaces and \cs{scan_stop:} tokens until
+%   we reach a non-expandable token. At that stage we re-insert the payload.
+%   To deal with the problem of |&| tokens, we have to put the align-safe
+%   group in the correct place.
+%    \begin{macrocode}
+\cs_new_protected:Npn \peek_remove_filler:n #1
+  {
+    \cs_set:Npn \@@_true_aux:w { \@@_remove_filler:w }
+    \cs_set:Npx \@@_false:w
+      {
+        \exp_not:N \group_align_safe_end:
+        \exp_not:n {#1}
+      }
+    \group_align_safe_begin:
+    \@@_remove_filler:w
+  }
+\cs_new_protected:Npn \@@_remove_filler:w
+  {
+    \exp_after:wN \peek_after:Nw \exp_after:wN \@@_remove_filler:
+    \exp:w \exp_end_continue_f:w
+  }
+\cs_new_protected:Npn \@@_remove_filler:
+  {
+    \if_catcode:w \exp_not:N \l_peek_token \c_space_token
+      \exp_after:wN \@@_true_remove:w
+    \else:
+      \if_meaning:w \l_peek_token \scan_stop:
+        \exp_after:wN \exp_after:wN \exp_after:wN
+          \@@_true_remove:w
+      \else:
+        \exp_after:wN \exp_after:wN \exp_after:wN
+          \@@_remove_filler_expand:w
+      \fi:
+    \fi:
+  }
```

Comment:
Maybe add a comment specifying that this nesting of conditionals is safe even if `\l_peek_token` is a TeX conditional (or an outer macro), because `\l_peek_token` is only skipped over if it is a space.